frontend/public/components/service-monitor.jsx

```diff
@@ -34,93 +52,130 @@ const serviceSelectorLinks = ({ spec }) => {
+      const cell = rowCells[id]?.cell || DASH;
+      const props = rowCells[id]?.props || undefined;
+      return {
+        id,
+        props,
+        cell,
+      };
+    });
+  });
 };
 
-const ServiceMonitorTableHeader = () => {
-  return [
-    {
-      title: 'Name',
-      sortField: 'metadata.name',
-      transforms: [sortable],
-      props: { className: tableColumnClasses[0] },
-    },
-    {
-      title: 'Namespace',
-      sortField: 'metadata.namespace',
-      transforms: [sortable],
-      props: { className: tableColumnClasses[1] },
-    },
-    {
-      title: 'Service Selector',
-      sortField: 'spec.selector',
-      transforms: [sortable],
-      props: { className: tableColumnClasses[2] },
-    },
-    {
-      title: 'Monitoring Namespace',
-      sortField: 'spec.namespaceSelector',
-      transforms: [sortable],
-      props: { className: tableColumnClasses[3] },
-    },
-    {
-      title: '',
-      props: { className: tableColumnClasses[4] },
-    },
-  ];
+const useServiceMonitorColumns = () => {
+  const { t } = useTranslation();
+  return React.useMemo(
+    () => [
+      {
+        title: t('public~Name'),
+        id: serviceMonitorTableColumnInfo[0].id,
+        sort: 'metadata.name',
+        props: {
+          ...cellIsStickyProps,
+          modifier: 'nowrap',
+        },
+      },
+      {
+        title: t('public~Namespace'),
+        id: serviceMonitorTableColumnInfo[1].id,
+        sort: 'metadata.namespace',
+        props: {
+          modifier: 'nowrap',
+        },
+      },
+      {
+        title: t('public~Service Selector'),
+        id: serviceMonitorTableColumnInfo[2].id,
+        sort: 'spec.selector',
+        props: {
+          modifier: 'nowrap',
+          width: 25,
+        },
+      },
+      {
+        title: t('public~Monitoring Namespace'),
+        id: serviceMonitorTableColumnInfo[3].id,
+        sort: 'spec.namespaceSelector',
+        props: {
+          modifier: 'nowrap',
+        },
+      },
+      {
+        title: '',
+        id: serviceMonitorTableColumnInfo[4].id,
+        props: {
+          ...cellIsStickyProps,
+        },
+      },
+    ],
+    [t],
+  );
 };
-ServiceMonitorTableHeader.displayName = 'ServiceMonitorTableHeader';
 
-export const ServiceMonitorsList = (props) => (
-  <Table
-    {...props}
-    aria-label="Service Monitors"
-    Header={ServiceMonitorTableHeader}
-    Row={ServiceMonitorTableRow}
-    virtualize
-  />
-);
+export const ServiceMonitorsList = (props) => {
+  const { data, loaded } = props;
+  const columns = useServiceMonitorColumns();
+
+  return (
+    <React.Suspense fallback={<LoadingBox />}>
+      <ConsoleDataView
+        {...props}
+        data={data}
+        loaded={loaded}
+        label={ServiceMonitorModel.labelPlural}
+        columns={columns}
+        initialFilters={initialFiltersDefault}
+        getDataViewRows={(dvData, dvColumns) => getServiceMonitorDataViewRows(dvData, dvColumns)}
```

Comment:
Nit - could be done directly as `getDataViewRows={getServiceMonitorDataViewRows}`